com.unity.netcode.gameobjects/Runtime/Serialization/SerializationTypeTable.cs

```diff
@@ -0,0 +1,449 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Unity.Netcode
+{
+    /// <summary>
+    /// Registry for telling FastBufferWriter and FastBufferReader how to read types when passed to
+    /// WriteObject and ReadObject, as well as telling BytePacker and ByteUnpacker how to do it when passed to
+    /// WriteObjectPacked and ReadObjectPacked.
+    ///
+    /// These object-based serialization functions shouldn't be used if at all possible, but if they're required,
+    /// and you need to serialize a type that's not natively supported, you can register it with the dictionaries here:
+    ///
+    /// Serializers and Deserializers for FastBufferWriter and FasteBufferReader
+    /// SerializersPacked and DeserializersPacked for BytePacker and ByteUnpacker
+    /// </summary>
+    public static class SerializationTypeTable
```

Comment:
so we're basically saying you can have completely separate serializers and deserializers for the same type? and even completely separate packed and non-packed serializers and deserializers? (in total 2 * 2 = 4 tables per type)

Reply:
Honestly, support for serializing `object` is only here because I don't know 100% for sure that I can kill it. I'll know when the IMessage stuff is done... And I'm hoping when I get there, I can just delete this file along with `ReadObject` and `WriteObject` entirely.